### PR TITLE
BUG: `from_dlpack`: fix default device

### PR DESCRIPTION
**Pull request by lucascolley**
closes gh-99

### CODE REVIEW

**Comment by lucascolley** [comment]
I can confirm that this unblocks SciPy

File: array_api_strict/tests/test_creation_functions.py [hunk 1]
```diff
@@ -236,3 +236,9 @@ def from_dlpack_2023_12(api_version):
     pytest.raises(exception, lambda: from_dlpack(capsule, copy=False))
     pytest.raises(exception, lambda: from_dlpack(capsule, copy=True))
     pytest.raises(exception, lambda: from_dlpack(capsule, copy=None))
+
+
+def test_from_dlpack_default_device():
+    x = asarray([1, 2, 3])
+    y = from_dlpack(x)
+    assert x.device is y.device
```
Comment (betatim) [inline array_api_strict/tests/test_creation_functions.py:244]: I think this should be
```suggestion
    assert x.device == y.device
```
because you can create `Device()` instances as you want and `Device` isn't a singleton or something like that. So `xp.Device(device="device2") is xp.Device("device2")` is `False`.

I think in this case the test passes because the instance for the default device is reused, but that seems like a "happy accident"/implementation detail?

Tl;DR: there is AFAIK no downside to `==` and it will always work

**Review by betatim — COMMENTED** [review]
One nitpick about `is` vs `==`.

We could add an assert to check that the device of `x` is actually the default device?

We could add a non regression test that uses a numpy array as argument for `from_dlpack`?

Otherwise looks good to me

**Comment by lucascolley** [comment]
thanks Tim, addressed

**Review by betatim — APPROVED** [review]

**Comment by betatim** [comment]
Thanks for the fix and the test!
